adapters/handlers/rest/authz/handlers_authz.go

```diff
@@ -76,6 +76,28 @@ func SetupHandlers(api *operations.WeaviateAPI, controller authorization.Control
 	api.AuthzGetRolesForOwnUserHandler = authz.GetRolesForOwnUserHandlerFunc(h.getRolesForOwnUser)
 }
 
+func (h *authZHandlers) AuthorizeRoleScopes(principal *models.Principal, originalVerb string, policies []authorization.Policy, roleName string) error {
+	// Check if user has full role management permissions
+	var err error
+	if err = h.authorizer.Authorize(principal, originalVerb, authorization.Roles(roleName)...); err == nil {
+		return nil
+	}
+
+	// Check if user can manage roles with matching permissions
+	if err = h.authorizer.Authorize(principal, authorization.ROLE_SCOPE_MATCH, authorization.Roles(roleName)...); err == nil {
```

Comment:
Shouldn't we check for `all` first? In case the user can do everything?

eg I think we should do

```
if err = h.authorizer.Authorize(principal, authorization.ROLE_SCOPE_ALL, authorization.Roles(roleName)...); err == nil {
   return //nothing to check
if err = h.authorizer.Authorize(principal, authorization.ROLE_SCOPE_MATCH, authorization.Roles(roleName)...); err == nil {
   ..existing code...
```

There are also no tests for this case

Reply:
the idea, it's to not check for all, but instead check for the requested permission 
we set for 
- all => C | R | U | D 
- match => MATCH

if the user has ALL any crud op shall be passing, but if the user has MATCH only it will check for permissions tries to assign

Reply:
would that mean that we check each permission for `all` but all of them succeed?

or to make more explizit what my question is:
```
h.authorizer.Authorize(principal, authorization.ROLE_SCOPE_MATCH, authorization.Roles(roleName)...)
```
Does ⬆️ return an error or succeed when a user has `all`? My expectation is that it would fail, but then we would return this error that is just below:
```
	return fmt.Errorf("can only create roles with less or equal permissions as the current user: %w", err)
```